modules/mkdru_ding/mkdru_ding.module

```diff
@@ -0,0 +1,339 @@
+<?php
+
+/**
+ * @file
+ * MKDRU Ding module.
+ */
+
+include_once 'mkdru_ding.features.inc';
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function mkdru_ding_ctools_plugin_directory($module, $plugin) {
+  if ($module == 'ctools' && $plugin == 'content_types') {
+    return 'plugins/' . $plugin;
+  }
+}
+
+/**
+ * Implements hook_search_info().
+ */
+function mkdru_ding_search_info() {
+  return array(
+    'title' => t('Digital Resources'),
+    'path' => 'meta',
+    'conditions_callback' => 'mkdru_ding_search_conditions_callback',
+  );
+}
+
+/**
+ * Implements hook_search_page().
+ */
+function mkdru_ding_search_page($results) {
+  $output['prefix']['#markup'] = theme('mkdru_results');
+  $output['suffix']['#markup'] = '';
+  return $output;
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function mkdru_ding_block_info() {
+  $blocks['mkdru_ding_search_results'] = array(
+    'info' => t('mkdru ding search results container.'),
+    'cache' => DRUPAL_NO_CACHE,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function mkdru_ding_block_view($delta = '') {
+  $block = array();
+
+  switch ($delta) {
+    case 'mkdru_ding_search_results':
+      $block['subject'] = t('mkdru ding search results');
+      $block['content'] = '<div class="mkdru-pager"></div><div class="mkdru-counts"></div><p></p><div class="mkdru-result-list"></div><div class="mkdru-pager"></div><div class="mkdru-counts"></div>';
```

Comment:
html shouldn't be hardcoded - instead, put it in a template file, and make a theme function for it.